android/src/main/java/com/jaumard/smsautofill/SmsAutoFillPlugin.java

```diff
@@ -56,13 +57,11 @@ private SmsAutoFillPlugin(MethodChannel channel, Registrar registrar) {
 
             @Override
             public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
-                if (requestCode == PHONE_HINT_REQUEST && pendingHintResult != null) {
-                    if (resultCode == Activity.RESULT_OK && data != null) {
+                if (requestCode == PHONE_HINT_REQUEST) {
+                    if (resultCode == Activity.RESULT_OK) {
                         Credential credential = data.getParcelableExtra(Credential.EXTRA_KEY);
                         final String phoneNumber = credential.getId();
                         pendingHintResult.success(phoneNumber);
-                    } else {
-                        pendingHintResult.success(null);
```

Comment:
and by removing this the dart side will never get any answer so the future will wait forever and take resources

Reply:
You mean the dart side will always await response and can't keep going?

Reply:
exactly

Reply:
OK I will fix it.